pkgs/tools/filesystems/btrfs-snap/default.nix

```diff
@@ -0,0 +1,27 @@
+{ bash, btrfs-progs, fetchFromGitHub, makeWrapper, lib, stdenv }:
+stdenv.mkDerivation rec {
+  pname = "btrfs-snap";
+  version = "1.7.3";
+  src = fetchFromGitHub {
+    owner = "jf647";
+    repo = pname;
+    rev = "${version}";
+    sha256 = "sha256-SDzLjgNRuR9XpmcYCD9T10MLS+//+pWFGDiTAb8NiLQ=";
+  };
+  buildInputs = [ bash btrfs-progs ];
+  nativeBuildInputs = [ makeWrapper ];
+  dontConfigure = true;
+  dontBuild = true;
+  installPhase = ''
+    mkdir -p $out/bin
+    cp * $out/bin/
+    wrapProgram $out/bin/btrfs-snap --prefix PATH : ${lib.makeBinPath [ btrfs-progs ]}
```

Comment:
also needs coreutils, util-linux and so on. Please try to run in an empty environment to check:
`env -i result/bin/btrfs-snap`

Reply:
Ouch. The script worked as-is on my NixOS machine. Surprised those deps got pulled in without me declaring them. Fixed.